src/lsqecc/pauli_rotations/rotation.py

```diff
@@ -280,6 +280,57 @@ def from_list(pauli_ops: List[PauliOperator], rotation: Fraction) -> "PauliRotat
             r.change_single_op(i, op)
         return r
 
+    @staticmethod
+    def from_fractional_phase(
+        num_qubits: int, target_qubit: int, phase_type: PauliOperator, phase: Fraction
+    ):
+        return PauliRotation.from_list(
+            [PauliOperator.I] * target_qubit
+            + [phase_type]
+            + [PauliOperator.I] * (num_qubits - target_qubit - 1),
+            rotation=phase / 2,
```

Comment:
I thought we were using `P_phi = exp(-i*P*phi)` as the convention from Litinski's convention, not the `P_phi = exp(-i*P*phi*0.5)`

Reply:
Phase is the argument of the `rz` gate, my understanding was that the argument is the relative phase to the `|1>` basis state, so `rz theta` = `exp(-i*pi*theta/2)` ... but this is definitely a subtle issue and a nice catch.

So what we are doing now is this:
```
Input Circuit:
     ┌─────────┐
q_0: ┤ Rz(π/8) ├
     └─────────┘
Circuit as Pauli rotations:
 q0---|Z|
pi*  1/16
```

Do you think it's correct?

Reply:
In qasm, the `theta/2` definition is used and internal in the `PauliRotation` class, we use the `theta`. This circuit is performing the correct operation, but just that generally `theta/2` definition is used. Line 291 takes care of the convention to maintain correctness of calculation. However, the end user might be confused, because he input `pi/8` but saw `pi/16` when the circuit is presented as Pauli Rotations

Reply:
Ok ... well it seems that the standard is to keep `theta` in the Pauli formalism ... but I do see how this would be confusing to users

Reply:
Added coments in various places to try make this issue more evident and what to do clearer, but I don't think we want to change our Pauli formalism (which matches Litinski's and [others's](https://arxiv.org/pdf/2109.02746.pdf)) based on what qiskit's and gridsynth's rz convention is